source/SAMRAI/tbox/AllocatorDatabase.h

```diff
@@ -23,21 +23,57 @@
 namespace SAMRAI {
 namespace tbox {
 
+/*!
+ * @brief Singleton class holding allocators for common SAMRAI operation
+ *
+ * This class provides access to Umpire allocators that are used to allocate
+ * data for specific operations that occur during runs of applications
+ * using SAMRAI.  The main intent is to support coordination of data
+ * allocations on the host and device when running with cuda-based GPU
+ * features enabled.  When not running with GPUs, these allocators will
+ * default to do regular allocations of CPU memory.
+ *
+ * Allocators defined here are:
+ *
+ * Device pool--A pool of memory on the device that can be used for
+ * temporary data that is created inside of kernels running on the GPU.
+ *
+ * Stream allocator--Allocator for pinned memory for MPI buffers used
+ * in communications launched by tbox::Schedule, most notably during the
+ * execution of refine and coarsen schedules.
+ *
+ * Tag allocator--Allocator for memory for the tag data object created and
+ * owned by GriddingAlgorithm and provided to applications.
```

Comment:
I think we could add something like the following:

These allocators can be overriden by creating Umpire allocators with the appropriate name prior to initializing SAMRAI. The names are `samrai::temporary_data_allocator`, `samrai::stream_allocator`, and `samrai::tag_allocator`. Please see the Umpire documentation for details on how to create new Allocators.

Reply:
@davidbeckingsale Sorry I left this hanging for a little while.